spec/auto_scaling_instance_spec.rb

```diff
@@ -0,0 +1,187 @@
+require 'spec_helper'
+require 'vcr_helper'
+
+RSpec.describe OpenStax::Aws::AutoScalingInstance, vcr: VCR_OPTS do
+
+  REGION = 'us-east-2'
+
+  STACK_EXISTS = true
+
+  context "me" do
+    # When recordingthese specs, run them on an autoscaling instance within AWS (see
+    # bin/create_development_environment).  The created instance will have the proper
+    # permissions to make the `me` call using the instance credentials.  However, VCR
+    # isn't set up to filter out these credentials, so just run them with your own
+    # AWS keys, e.g.:
+    #
+    # $> VCR_OPTS_RECORD=all AWS_ACCESS_KEY_ID=foo AWS_SECRET_ACCESS_KEY=bar \
+    #      bundle exec rspec ./spec/auto_scaling_instance_spec.rb:21
+
+    it "makes an ASI" do
+      expect(described_class.me.lifecycle_state).to eq "InService"
+    end
+  end
+
+  context "terminate things" do
+
+    before(:context) {
```

Comment:
i get it...but all these befores, afters, arounds...make the test diffiicult to grok.  there are a lot for this test.

Reply:
I'll add some comments.  The `before(:context)` is to set up the AWS resources once and reuse them across specs.  The `around` and `before` with the logger are because you can only use instance vars in a `before`, and I like the `around` for cleanup.